gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/LocalRegion.java

```diff
@@ -14352,6 +14353,12 @@ private long getEntryOverhead(RegionEntry entry){
     Object key = entry.getRawKey();
     if (key != null) {
       entryOverhead += ReflectionSingleObjectSizer.INSTANCE.sizeof(key);
+    }else{
+      // first key.
+      Object firstKey = this.getRegionMap().keySet().iterator().next();
+      if(firstKey != null){
+        entryOverhead += ReflectionSingleObjectSizer.INSTANCE.sizeof(firstKey);
+      }
```

Comment:
We should not use ReflectionSingleObjectSizer directly anywhere (except special cases like LocalRegion overhead). Rather use CachedDeserializableFactory.calcMemSize because that will also work correctly for Sizeable objects (that can use ReflectionSingleObjectSizer internally if required) which may be deep and not work correctly with shallow reflection. At most use the overload to pass it an ObjectSizer based on RelectionSingleObjectSizer like:
```java
  static final com.gemstone.gemfire.cache.util.ObjectSizer GEM_SIZER =
      o -> (int)RelectionSingleObjectSizer.sizeof(o);
```
I have already made this change in MemoryAnalytics ObjectSizer used by TableMBeans.